src/Sentry/SentryGraphQLHttpFailedRequestHandler.cs

```diff
@@ -24,15 +25,24 @@ protected internal override void DoEnsureSuccessfulResponse([NotNull] HttpReques
         JsonElement? json = null;
         try
         {
-            json = GraphQLContentExtractor.ExtractResponseContentAsync(response, _options).Result;
+            json = GraphQLContentExtractor.ExtractResponseContentAsync(response, _options).GetAwaiter().GetResult();
             if (json is { } jsonElement)
             {
                 if (jsonElement.TryGetProperty("errors", out var errorsElement))
                 {
                     // We just show the first error... maybe there's a better way to do this when multiple errors exist.
                     // We should check what the Java code is doing.
                     var errorMessage = errorsElement[0].GetProperty("message").GetString() ?? "GraphQL Error";
-                    throw new GraphQLHttpRequestException(errorMessage);
+                    var exception = new GraphQLHttpRequestException(errorMessage);
+
+#if NET5_0_OR_GREATER
+                    // Add a full stack trace into the exception to improve Issue grouping,
+                    // see https://github.com/getsentry/sentry-dotnet/issues/3582
+                    ExceptionDispatchInfo.Throw(ExceptionDispatchInfo.SetCurrentStackTrace(exception));
```

Comment:
So I think I'm getting confused now. 

How is this:
```
var exception = new GraphQLHttpRequestException(errorMessage);
ExceptionDispatchInfo.Throw(ExceptionDispatchInfo.SetCurrentStackTrace(exception));
```

... different from this?
```
var exception = new GraphQLHttpRequestException(errorMessage);
throw exception;
```

I get that the behaviour for `ExceptionDispatchInfo.Throw` is different to `throw` when we're talking about exceptions that have been thrown elsewhere in the code and have a stack trace to preserve... in this case we're creating the exception and throwing it on the very next line though... so what is the value of `ExceptionDispatchInfo.Throw` vs a plain old `throw` in this scenario? Aren't the stack traces going to be almost identical?

Reply:
It is different, I know it's not obvious--try it via the unit test and debugging, if you're interested. The one test should fail if you just throw

Reply:
That's OK - this problem space turns us all on our heads at some point. 

Thanks heaps for looking into this @ericsampson and should we close this PR then?